compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala

```diff
@@ -745,7 +746,9 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext:
       def unapply(tree: Block): Option[(List[ValDef], Term)] = tree match {
         case Block((ddef @ DefDef(_, _, params :: Nil, _, Some(body))) :: Nil, Closure(meth, _))
         if ddef.symbol == meth.symbol =>
-          Some((params, body))
+          val cleanParams = params.map(_.changeOwner(meth.symbol,ctx.owner))
+          val cleanBody = body.changeOwner(meth.symbol,ctx.owner)
```

Comment:
We cannot rely on this body being used inside `ctx.owner`. We need to change adapt it when we construct the `Lambda`. We have some logic that knows how to do this. I created an alternative fix that does just that in #10142.

Reply:
This fix was helped me understand the source of the issue. Thank you.

Reply:
ok, thanks